llvm/lib/Analysis/DXILResource.cpp

```diff
@@ -775,6 +778,42 @@ void DXILResourceMap::populate(Module &M, DXILResourceTypeMap &DRTM) {
     // Adjust the resource binding to use the next ID.
     RI.setBindingID(NextID++);
   }
+
```

Comment:
I wonder if it would be good to move this and the code above into separate methods like `populateResourceInfos` and `analyzeCounterDirections` that would be called from `populate`. I believe there is also going to be a similar part for analyzing globally coherent resources.